### PR TITLE
Add Direction Provider

### PR DESCRIPTION
**Pull request by mgjeffries**
The directions provider exports two functions, one for getting directions via a fetch call, and one for using the directions that are provided.

### CODE REVIEW

File: scripts/directions/DirectionProvider.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      directionsList = responseContent.paths["0"].instructions.map( instructionObj => {
+        return instructionObj.text
+      } )
+    })
+} 
+
+// a function that creates a list of latitude longitude pairs from an itenerary object
+const getLatLngPairs = itineraryObj => {
+  return getLatLngPair( startingCity+"+"+startingState ) 
+    .then( () => getLatLngPair(itineraryObj.parkInfo.parkCity+"+"+itineraryObj.parkInfo.parkState+"+"+defaultCountry)) //example of gettting latLongs for multiple cities
```
Comment (skratz17) [inline scripts/directions/DirectionProvider.js]: Is this comment necessary? Or was this something you had while testing that just never got deleted?

**Comment by skratz17** [comment]
It seems like we want to wipe out the `LatLongPairs` array maybe right after we build the `latLongString` string - as it is now it seems like if the user clicks "Get Directions" for multiple itineraries, this `LatLongPairs` array is just gonna keep getting bigger and bigger as it is never getting cleared out from the previous time the user got directions.

File: scripts/directions/DirectionProvider.js (hunk 1 above)
Comment (mgjeffries) [reply scripts/directions/DirectionProvider.js]: Not necessary, since the code is clear enough, I'll take it out.

**Review by skratz17 — APPROVED** [review]
Looks good to me 👍

**Review by Raparker88 — APPROVED** [review]

**Review by skratz17 — APPROVED** [review]

**Pull request merged**